packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1257,8 +1319,25 @@ export class TransactionController extends BaseController<
         delete txParams.gasPrice;
       }
 
+      if (!this.beforePublish(transactionMeta)) {
+        transactionMeta.status = TransactionStatus.approved;
+        this.updateTransaction(
+          transactionMeta,
+          'TransactionController#approveTransaction - Transaction approved',
+        );
+      }
+
       const unsignedEthTx = this.prepareUnsignedEthTx(txParams);
-      const signedTx = await this.sign(unsignedEthTx, from);
+      const signedTx = await this.sign(
+        unsignedEthTx,
+        from,
+        ...this.getAdditionalSignArguments(transactionMeta),
+      );
+      // In certain cases, the responsibility to publish the transaction lies with the custodian
```

Comment:
Do we want to avoid coupling to the hook domains even through comments, to ensure maximum clarity?